paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc

```diff
@@ -100,34 +114,45 @@ class MatMulFactory {
     const memory::dim M = mat_dim_x.height_;
     const memory::dim N = mat_dim_y.width_;
     const memory::dim K = mat_dim_x.width_;
-    return {BS, M, N, K};
+
+    batch_size_ = 1;
+    auto b = BS;
+    if (BS > 1 && IsOutputFused(ctx)) {
+      batch_size_ = ctx.Input<Tensor>("X")->dims()[0];
+      b = BS / batch_size_;
+    }
+    memory::dims x_dims = {b, M, K};
+    memory::dims y_dims = {b, K, N};
+    memory::dims out_dims = {b, M, N};
+
+    size_t x_size = b * M * K * sizeof(XT);
+    size_t y_size = b * K * N * sizeof(YT);
+    size_t out_size = b * M * N * sizeof(OT);
+    offsets_ = {x_size, y_size, out_size};
+
+    // Translate transA and transB
+    memory::dims strides_x = !ctx.Attr<bool>("transpose_X")
+                                 ? memory::dims{M * K, K, 1}
+                                 : memory::dims{M * K, 1, M};
+    memory::dims strides_y = !ctx.Attr<bool>("transpose_Y")
+                                 ? memory::dims{N * K, N, 1}
+                                 : memory::dims{N * K, 1, K};
+    memory::dims out_strides = memory::dims{M * N, N, 1};
+
+    correctStridesWhenOutputFused(ctx, N, b, &out_strides);
```

Comment:
Please consider renaming `correctStridesWhenOutputFused` to `CorrectStridesWhenOutputFused`

Reply:
done